xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/getdocuments.vm

```diff
@@ -80,6 +80,10 @@ $response.setContentType("application/json")
 ## ORDER
 ##
 #set ($order = "$!request.sort")
+#if (!$regextool.compile('^(doc\.)?\w+$').matcher($order).matches())
+  { "error" : "invalid sort parameter" }
+  #stop ("invalid sort parameter [$order] found")
```

Comment:
I'm wondering if logging this input parameter is a good idea (I think this is a debug log), this might allow [log injection](https://owasp.org/www-community/attacks/Log_Injection) unless we already have some code to prevent that. I don't know if we care about this as this is also just a debug log.

Reply:
> if this indeed goes to the debug log

From what I see https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-core/xwiki-commons-velocity/src/main/java/org/xwiki/velocity/internal/DefaultVelocityEngine.java#L286 it seems we don't even log it at all.

Reply:
Sorry, I missed the discussion in the chat, was otp. Should I write a loop over all values of the `sort` request parameter?
(Also I noticed that my check will fail if no `sort` parameter is present, ouch, fixing)